gapir/cc/grpc_replay_service.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "grpc_replay_service.h"
+
+#include <grpc++/grpc++.h>
+#include <memory>
+
+#include "core/cc/log.h"
+#include "gapir/replay_service/service.grpc.pb.h"
+#include "gapir/replay_service/service.pb.h"
+#include "gapis/service/severity/severity.pb.h"
+
+namespace gapir {
+
+std::unique_ptr<ReplayService::Payload> GrpcReplayService::getPayload() {
+  // Send a replay response with payload request
+  replay_service::ReplayResponse res;
+  res.set_allocated_payload_request(new replay_service::PayloadRequest());
+  mGrpcStream->Write(res);
+  std::unique_ptr<replay_service::ReplayRequest> req(
+      new replay_service::ReplayRequest());
+  if (!mGrpcStream->Read(req.get())) {
+    return nullptr;
+  }
+  if (req->req_case() != replay_service::ReplayRequest::kPayload) {
+    return nullptr;
+  }
+  return std::unique_ptr<ReplayService::Payload>(new ReplayService::Payload(
```

Comment:
This looks like a good place for make_unique, (if it actually exists in all of our compilers)